src/test/versionbits_tests.cpp

```diff
@@ -284,100 +259,67 @@ BOOST_AUTO_TEST_CASE(versionbits_params)
 BOOST_AUTO_TEST_CASE(versionbits_computeblockversion)
 {
     // Check that ComputeBlockVersion will set the appropriate bit correctly
-    // on mainnet.
-    const auto chainParams = CreateChainParams(*m_node.args, CBaseChainParams::MAIN);
+    const auto period = CreateChainParams(*m_node.args, CBaseChainParams::REGTEST)->GetConsensus().nMinerConfirmationWindow;
+    gArgs.ForceSetArg("-vbparams", strprintf("testdummy:@%s:@%s", period, period * 2));
+    const auto chainParams = CreateChainParams(*m_node.args, CBaseChainParams::REGTEST);
```

Comment:
I believe this change was done so that custom activation parameters could be used. Specifically, since mainnet start and timeout heights are in the hundreds of thousands of blocks, the test would just be wasting time building up a chain to reach that those heights. With regtest, it can use a way lower height so the test runs in a timely fashion.

Reply:
I guess the better question is: why did it use mainnet in the past?

Reply:
I guess it was convenient and doesn't really matter when the time can be faked?

Reply:
I figured the theory was "if you're going to test anything, test what's going to happen on mainnet". Since there's no proof of work on mainnet, it might be reasonable to test with real heights -- it's just building a skiplist with half a million entries to get to the start height, and going from there?

Reply:
Oh, no -- the reason was that it is possible to test "TESTDUMMY" activation on mainnet with MTP because you can fake the block time to be prior to the the genesis block's timestamp; but you can't do that with height because you obviously can't have a block height prior to NEVER_ACTIVE ie -2.